### ISSUE TITLE
New CNCI image creation has broken ciao builds are broken

### ISSUE THREAD

**Issue by markdryan** [ciao-project/ciao#1586]
There are two separate problems caused by https://github.com/ciao-project/ciao/commit/c84b55bed9f9bd4ca20f23c3b59e5fb98136b9b5

1. There's an issue with the latest clear linux release, in that the image is actually missing from the latest folder.  I have a patch (https://github.com/ciao-project/ciao/pull/1585) that introduces a hardcoded faillback, but this doesn't fix the builds, see next item.
2. The builds, often fail with the following error

Error provisioning system as master: Error creating CNCI image: Error preparing image: Error unmounting image: Error running: [sudo umount /tmp/cnci-mount011821163]: exit status 32

This happens when we try to unmount the CNCI image after having installed the various bundles.  This triggers a whole pile of failures in ciao-deploy which leads to a cluster that does not work.  The problem seems to be caused by the bundle-add command.  When I comment this out I'm able to unmount the temporary folder.  Also, adding a swupd verify after the swupd bundle-add seems to fix the problem, but this is just a timing issue I guess.

Here's an example build that fails

https://semaphoreci.com/kaccardi/ciao/branches/pull-request-1585/builds/1

**Comment by markdryan** [comment]
I'm going to revert c84b55b for now as it breaks all our builds.

**Comment by kaccardi** [comment]
@markdryan for issue 1) - it turns out "latest" isn't what we thought and we should stop using it to determine which clear version to download. Instead, the last good version of the image we want will always be located in "download.clearlinux.org/image", and there should be only one of them. I rewrote the original patch to stop using "latest". (see #1593 ). For 2 - I was suspicious that something going on in a defer statement was keeping the chroot busy, so in a couple spots I modified our use of exec.Run() to use exec.CombinedOutput() instead so that I could see the output of certain things. For some reason with this change I can no longer reproduce the problem. Any idea why just switching to CombinedOutput would impact anything? I'm not even outputting the output I gather since it only displays during error path, and there are no errors.
